### ISSUE TITLE
pods get recreated daily

### ISSUE THREAD

**Issue by mkreuzmayr** [kube-hetzner/terraform-hcloud-kube-hetzner#463]
I don´t know if this has to do someting with kube-hetzner in particular but I noticed the following today and I cannot think of what happened. Somehow every day all of my pods get recreated with the exception of some in the pods kube-system namespace, they only get restarted.

![kube](https://user-images.githubusercontent.com/20108212/208396093-4ec47538-8743-42d2-ad8d-871d9f092596.jpg)

I have got three nodes and they are all up since 25d. My hetzner control panel also does not show any activities concerning node creation/deletion. It only shows service updates and volume/attach detaches.

![events](https://user-images.githubusercontent.com/20108212/208400012-2e5a0c8b-7c0d-4ec4-8737-265a0ee380f7.jpg)

I already inspected logs and events in kubernetes or if I have any jobs or tasks that could lead to that.

**Comment by mkreuzmayr** [comment]
Ok, that makes sense. If i have multiple replicas of my pods, will the service be available while updating or is there a short downtime?

**Comment by otavio** [comment]
This also concerns me. We are moving our services to this and the downtime is a bad thing for us.

**Comment by maaft** [comment]
I have the same issue, but in particular with Hashicorp Vault: After being restarted, the vault gets automatically sealed. And for security purposes, the unseal process must be manual.

Meaning, our corporate secret store (vault) would required manual intervention every day to stay functional.

Any idea what I can do instead? Could we do manual, scheduled upgrades of MicroOS instead? If so, how?

**Comment by mrhein** [comment]
from architecture perspective you can solve it with a management cluster running rockylinux or simlar with your vault harbor and so on and your application cluster in a more updated way the problem is than you have 2 setups to manage and a more vulnerable platform 

maybe hetzner will plugin one of this YubiHSM 2 if we ask them (atleast for metal server), because vault says...
> Vault integrates with the YubiHSM 2 via PKS#11 for customers who need hardware backed security. The integration supports auto unseal, cryptographic key generation and management, & external entropy.

so or so the kured config should be edited
https://kured.dev/docs/configuration/#setting-a-schedule
https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/master/kustomize/kured.yaml#L19

that should help you out maybe update times only 3 days in row in middle of the week 9 to 11 am

p.s. @maaft there are only one solution on the market that is great and that ist zalando => postgres opertor which spawns patroni => spilo for you

**Comment by mysticaltech** [comment]
@mrhein Exactly. This is by design. When an update comes, microOS applies in a new snapshot, then signals it to kured, which in turn drains the node and reboots to apply the new snapshot. If you are running in HA, >= 3 control planes, and > 2 agents, you should have 0 downtime.

Now @mkreuzmayr @maaft Note that you can disable this behavior if you choose so, there are variables for that, also see the manual way in the readme.

**Comment by mysticaltech** [comment]
Moving this to a discussion.
